File: terceiro_semestre/tecnicas/LEE04/ex005/churrasco.cpp
#include <iostream>
#include <set>
#include <string>

using namespace std;

int main() {
    int A;
    cin >> A;

    // Conjunto para armazenar todos os itens únicos do churrasco
    set<string> itens_churrasco;

    // Processamento dos itens de cada aluno
    for (int i = 0; i < A; ++i) {
        int P;
        cin >> P;
        cin.ignore(); // Ignora o newline após o número de itens

        for (int j = 0; j < P; ++j) {
            string item;
            getline(cin, item);

            // Verificação e adição do item no conjunto
            if (itens_churrasco.find(item) != itens_churrasco.end()) {
                cout << item << " ja tem" << endl;
            } else {
                cout << "adicionando " << item << endl;
                itens_churrasco.insert(item);
            }
        }
    }

    // Impressão dos itens finais em ordem alfabética
    cout << "Itens do churrasco:" << endl;
    for (const auto& item : itens_churrasco) {
        cout << item << endl;
    }

    return 0;
}

File: terceiro_semestre/tecnicas/LEE04_py/ex005/churrasco.py
A = int(input().strip())

lista_churrasco = set()

for i in range(A):
    P = int(input().strip())
    
    for i in range(P):
        produto = input().strip()
        if produto in lista_churrasco:
            print(f"{produto} ja tem")
        else:
            print(f"adicionando {produto}")
            lista_churrasco.add(produto)

print("Itens do churrasco:")
for produto in sorted(lista_churrasco):
    print(produto)
